pulp_ansible/tests/functional/api/collection/v3/test_sync.py

```diff
@@ -160,3 +162,117 @@ def test_v3_sync(self):
             original_content.data[0].highest_version["version"],
             COLLECTION_VERSION,
         )
+
+
+@unittest.skipUnless(
+    "AUTOMATION_HUB_TOKEN_AUTH" in os.environ,
+    "'AUTOMATION_HUB_TOKEN_AUTH' env var is not defined",
+)
+class AutomationHubV3SyncCase(unittest.TestCase, CreateRepoAndSyncMixin):
+    """Test syncing from Pulp to Pulp."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = gen_ansible_client()
+        cls.repo_api = RepositoriesAnsibleApi(cls.client)
+        cls.remote_collection_api = RemotesCollectionApi(cls.client)
+        cls.distributions_api = DistributionsAnsibleApi(cls.client)
+        cls.collections_api = PulpAnsibleGalaxyApiCollectionsApi(cls.client)
+        cls.cv_api = ContentCollectionVersionsApi(cls.client)
+
+    def test_sync_with_token_from_automation_hub(self):
+        """Test whether we can sync with an auth token from Automation Hub."""
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        aurl = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+        body = gen_ansible_remote(
+            url="https://cloud.redhat.com/api/automation-hub/",
+            requirements_file="collections:\n  - ansible.posix",
+            auth_url=aurl,
+            token=os.environ["AUTOMATION_HUB_TOKEN_AUTH"],
+            tls_validation=False,
+        )
+        remote = self.remote_collection_api.create(body)
+        self.addCleanup(self.remote_collection_api.delete, remote.pulp_href)
+
+        repo = self._create_repo_and_sync_with_remote(remote)
+
+        # Check content of both repos.
+        original_content = self.cv_api.list(repository_version=f"{repo.pulp_href}versions/1/")
+        self.assertTrue(len(original_content.results) >= 3)  # check that we have at least 3 results
+
+
+@unittest.skipUnless(
+    "CI_AUTOMATION_HUB_TOKEN_AUTH" in os.environ,
+    "'CI_AUTOMATION_HUB_TOKEN_AUTH' env var is not defined",
+)
+class AutomationHubCIV3SyncCase(unittest.TestCase, CreateRepoAndSyncMixin):
+    """Test syncing from Pulp to Pulp."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = gen_ansible_client()
+        cls.repo_api = RepositoriesAnsibleApi(cls.client)
+        cls.remote_collection_api = RemotesCollectionApi(cls.client)
+        cls.distributions_api = DistributionsAnsibleApi(cls.client)
+        cls.collections_api = PulpAnsibleGalaxyApiCollectionsApi(cls.client)
+        cls.cv_api = ContentCollectionVersionsApi(cls.client)
+        cls.aurl = (
+            "https://sso.qa.redhat.com/auth/realms/redhat-external/protocol/" "openid-connect/token"
+        )
```

Comment:
why this url is splitted?

Reply:
Ah I split it because it was too long and then black put it back. I'll fix it. Ty